### PR TITLE
Fixed download limit of 200 on the compounds page

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

This  fixes the download limit of 200 rows by replacing the bootstrap download button with a custom button.  I figured out how to do it inside the bootstrap toolbar itself.  Only tsv is supported.  See the discussion in the comments in the issue for details on the problem and the fix.

This still needs to be done for every template.  It represents just over a day's work, so if we want to go a different route, that's fine with me.

Details:

- Added a new template tag: `getDownloadQry`
- Added a download form and javascript to trigger it in the compound list page.
- Removed the export attributes and added a data-buttons attribute to put the custom button in the bootstrap toolbar.
- Added a method to `FormatGroup`: `getDownloadQry` that takes a format name and returns a dict with the download name and the root qry object.

## Affected Issues/Pull Requests

- Partially addresses #1038
- Merges into PR #1154

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/formats/dataformat_group.py [hunk 1]
```diff
@@ -774,6 +774,15 @@ def getDownloadQryList(self):
...
         return qry_list
 
+    def getDownloadQry(self, format):
+        """Returns a dict where the keys are name and json and the values are the format name and the json-stringified
+        qry object with the target format selected.
+        """
+        root_group = self.getRootGroup(format)
+        name = root_group["searches"][format]["name"]
+        qry = {"name": name, "json": json.dumps(root_group)}
+        return qry
```
Comment (hepcat72) [inline DataRepo/formats/dataformat_group.py:784]: C. This is a complement to `getDownloadQryList` above - when you only want one specific format search/download.

File: DataRepo/templates/DataRepo/compound_list.html [hunk 2]
```diff
@@ -20,8 +57,7 @@ <h4>List of Compounds</h4>
         data-show-columns="true"
         data-show-columns-toggle-all="true"
         data-show-fullscreen="true"
-        data-show-export="true"
-        data-export-types="['csv', 'txt', 'excel']"
+        data-buttons="buttonsFunction"
```
Comment (hepcat72) [inline DataRepo/templates/DataRepo/compound_list.html:60]: C. I didn't know what to name `buttonsFunction`, but that's what the name was in the bootstrap example for custom toolbar buttons.  It probably doesn't matter.

File: DataRepo/templatetags/customtags.py [hunk 3]
```diff
@@ -112,6 +112,12 @@ def getDownloadQrys():
     return basv.getDownloadQryList()
 
 
+@register.simple_tag
+def getDownloadQry(format):
+    basv = SearchGroup()
+    return basv.getDownloadQry(format)
```
Comment (hepcat72) [inline DataRepo/templatetags/customtags.py:118]: C. And this just provides access to `SearchGroup.getDownloadQry` from the template.

**Comment by lparsons** [comment]
@hepcat72 Not sure what is going on here, but this PR is closed and unmerged. I'll hold off on reviews since I'm guessing this is being rethought, but let me know if I should look at it.

**Comment by hepcat72** [comment]
> @hepcat72 Not sure what is going on here, but this PR is closed and unmerged. I'll hold off on reviews since I'm guessing this is being rethought, but let me know if I should look at it.

Learned a simpler fix in #1164. John reviewed it I think and it was merged. This one was abandoned.

**Pull request closed without merging**